Source/CarthageKit/Project.swift

```diff
@@ -684,50 +698,85 @@ public final class Project {
+		let rawDependencyURL = rootDirectoryURL.appendingPathComponent(dependency.project.relativePath, isDirectory: true)
 		let dependencyURL = rawDependencyURL.resolvingSymlinksInPath()
 		let dependencyCheckoutsURL = dependencyURL.appendingPathComponent(CarthageProjectCheckoutsPath, isDirectory: true).resolvingSymlinksInPath()
-		let subDependencyNames = subDependencies.map { $0.name }
 		let fileManager = FileManager.default
 
-		let symlinksProducer = SignalProducer<String, CarthageError>(subDependencyNames)
-			.filter { name in
-				let checkoutURL = rootCheckoutsURL.appendingPathComponent(name)
-				do {
-					return try checkoutURL.resourceValues(forKeys: [ .isDirectoryKey ]).isDirectory ?? false
-				} catch {
-					return false
-				}
-			}
-			.attemptMap { name -> Result<(), CarthageError> in
-				let dependencyCheckoutURL = dependencyCheckoutsURL.appendingPathComponent(name)
-				let subdirectoryPath = (CarthageProjectCheckoutsPath as NSString).appendingPathComponent(name)
-				let linkDestinationPath = relativeLinkDestinationForDependencyProject(dependency, subdirectory: subdirectoryPath)
-				do {
-					try fileManager.createSymbolicLink(atPath: dependencyCheckoutURL.carthage_path, withDestinationPath: linkDestinationPath)
-				} catch let error as NSError {
-					if !(error.domain == NSCocoaErrorDomain && error.code == NSFileWriteFileExistsError) {
-						return .failure(.writeFailed(dependencyCheckoutURL, error))
+		return self.dependencyProjects(for: dependency)
+			.zip(with: // file system objects which might conflict with symlinks
+				fileSystemObjects(for: dependency, withRepository: repositoryURL, at: CarthageProjectCheckoutsPath, flatMapTransform: { path in
+					let componentsRelativeToDirectoryURL = {
+						return URL(string: $0, relativeTo: self.directoryURL)?.standardizedFileURL.carthage_pathComponents
 					}
-				}
-				return .success()
-			}
+					if
+						let components = componentsRelativeToDirectoryURL(path),
+						let comparator = componentsRelativeToDirectoryURL(CarthageProjectCheckoutsPath),
+						Array(components.dropLast(1)) == comparator
+					{
+						return components.last!
+					} else {
+						return nil
+					}
+				})
+			)
+			.attemptMap { (dependencies: Set<ProjectIdentifier>, components: [String]) -> Result<(), CarthageError> in
+				let names = dependencies.filter { dependency in
+					// Filter out dependencies with names matching (case-insensitively) file system objects from git in `CarthageProjectCheckoutsPath`.
+					// Edge case warning on file system case-sensitivity. If a differently-cased file system object exists in git
+					// and is stored on a case-sensitive file system (like the Sierra preview of APFS), we currently preempt
+					// the non-conflicting symlink. Probably, nobody actually desires or needs the opposite behavior.
+					components.filter {
+						dependency.name.caseInsensitiveCompare($0) == .orderedSame
+					}.isEmpty
+				}.map { $0.name }
```

Comment:
Can you format this in the RAC style—where each operator is on an indented line?

```swift
let names = dependencies
    .filter {
        ...
    }
    .map { $0.name }
```